src/opnsense/mvc/app/library/OPNsense/Backup/Mailer.php

```diff
@@ -195,13 +195,11 @@ public function sendEmail($config, $confdata)
         $gpgPublicKeyFile = "key.asc";
         if ($gpgPublicKey != "") {
             file_put_contents($gpgPublicKeyFile, $gpgPublicKey);
+            proc_open('gpg --import < ' . $gpgPublicKeyFile);
         }
 
-
-        proc_open('gpg2 --import < ' . $gpgPublicKeyFile);
-
         proc_open('echo "' . $confdata . '"
-            | gpg2 --trust-model always --batch --yes --output backup --encrypt --recipient ' .
+            | gpg --trust-model always --batch --yes --output backup --encrypt --recipient ' .
```

Comment:
no, you should open the pipes and use them instead of the echo ... | part

Reply:
I think I fixed it now, thanks for the tips